addons/mail/static/src/components/message_list/message_list.xml

```diff
@@ -5,7 +5,7 @@
         <div class="o_MessageList" t-att-class="{ 'o-empty': threadView and threadView.messages.length === 0, 'o-has-message-selection': props.selectedMessageLocalId }" t-on-scroll="_onScroll">
             <t t-if="threadView">
                 <!-- No result messages -->
-                <t t-if="threadView.messages.length === 0">
+                <t t-if="threadView.threadCache.isEmpty">
```

Comment:
I prefer having field `nonEmptyMessages` than `isEmpty`.
The support of empty message is caused by clumsy support of message deletion.
I wouldn't increase complexity of code comprehension around the notion of "empty" message, especially when the mid-long term aim is to clean message deletion and no longer have this weird "empty" messages.

Reply:
To clarify, it's much more explicit to have `nonEmptyMessages` than `isEmpty`.
Eventually will remove notion of "is empty", and doing so would just be `nonEmptyMessages === messages`, so just a rename.